### PR TITLE
CFY-2010 added documentation on the autoheal built in workflow

### PR DESCRIPTION
**Pull request by iliapolo**

### CODE REVIEW

File: source/guide/3.2/reference-builtin-workflows.md [hunk 1]
```diff
@@ -106,3 +106,31 @@ For each of the remaining node instances:
 <sub>
 1. Note that the dependency may be indirect, e.g. in a case where instance A is dependent on instance B, which is in turn dependent on instance C, and only B was filtered out, instance A's operation execution will still only happen after instance C's operation execution.
 </sub>
+
+# Auto-heal
+
+**Workflow name:** *autoheal*
+
+**Workflow description:** The workflow for re-installing a sub graph of the blueprint.
```
Comment (m1keil) [inline source/guide/3.2/reference-builtin-workflows.md]: what's a graph? or a sub graph? needs a reference

File: source/guide/3.2/reference-builtin-workflows.md [hunk 2]
```diff
@@ -106,3 +106,31 @@ For each of the remaining node instances:
...
+
+**Workflow parameters:**
+
+  - *node_id*: The id of the failed node instance.
+
+**Workflow high-level pseudo-code:**
+
+  1. Retrieve the host id of the failing node instance.
```
Comment (m1keil) [inline source/guide/3.2/reference-builtin-workflows.md]: failed node (it's actual state right?)

File: source/guide/3.2/reference-builtin-workflows.md [hunk 3]
```diff
@@ -106,3 +106,31 @@ For each of the remaining node instances:
...
+      - Execute uninstall lifecycle operations (`stop`, `delete`) on the host and all it's contained nodes.
+      - Execute uninstall relationship lifecycle operations (`unlink`) for all affected relationships.
+
+  4. Install the sub-graph:
+
+      - Execute install lifecycle operations (`create`, `configure`, `start`) on the host and all it's contained nodes.
+      - Execute install relationship lifecycle operations (`preconfigure`, `postconfigure`, `establish`) for all affected relationships.
+
+<sub>
+1. Note that effectively, all nodes that are contained inside the host of the failing node, are considered failed as well and will be re-installed.
```
Comment (m1keil) [inline source/guide/3.2/reference-builtin-workflows.md]: (1) isn't assigned anywhere? maybe should be converted to a note or asterisk?

File: source/guide/3.2/reference-builtin-workflows.md [hunk 4]
```diff
@@ -106,3 +106,109 @@ For each of the remaining node instances:
...
+
+**Workflow description:** The workflow for re-installing specific blueprint nodes.
+
+**Workflow parameters:**
+
+  - *node_id*: The id of the failed node instance.
+
+**Workflow high-level pseudo-code:**
+
+  1. Retrieve the host of the failed node.
```
Comment (idanmo) [inline source/guide/3.2/reference-builtin-workflows.md]: host -> Compute node

**Pull request merged**